src/mean.jl

```diff
@@ -46,13 +46,51 @@ julia> mean(:api00, bclus1)
 ```
 """
 function mean(x::Symbol, design::ReplicateDesign)
-    X = mean(design.data[!, x], weights(design.data[!, design.weights]))
-    Xt = [
-        mean(design.data[!, x], weights(design.data[!, "replicate_"*string(i)])) for
-        i = 1:design.replicates
-    ]
-    variance = sum((Xt .- X) .^ 2) / design.replicates
-    DataFrame(mean = X, SE = sqrt(variance))
+    if design.type == "bootstrap"
+        θ̂ = mean(design.data[!, x], weights(design.data[!, design.weights]))
+        θ̂t = [
+            mean(design.data[!, x], weights(design.data[!, "replicate_"*string(i)])) for
+            i = 1:design.replicates
+        ]
+        variance = sum((θ̂t .- θ̂) .^ 2) / design.replicates
+    
+    elseif design.type == "jackknife"
+        
+        # The estimate based on original sampling weight vector
+        θ̂ = mean(design.data[!, x], weights(design.data[!, design.weights]))
+        
+        # Find number of psus (nh) in each strata, used inside loop
+        stratified_gdf = groupby(design.data, design.strata)
+        nh = Dict{String, Int}()
+        for (key, subgroup) in pairs(stratified_gdf)
+            nh[key[design.strata]] = length(unique(subgroup[!, design.cluster]))
+        end
```

Comment:
Hi @smishr . Regarding lines 62-67: we will need the number of psus in each strata in almost all variance estimation methods, right? So wouldn't it be a good idea to have these as a property of the struct for reuse later?

Reply:
yes ive been thinking of this. Dont need it for ALL methods per se, but if its cheaper to store than to calculate on fly then lets store them as a Dict from Symbol/String to Int